src/components/AddComment.jsx

```diff
@@ -0,0 +1,63 @@
+import { useContext, useState } from "react";
+import { postComment } from "../../Utils/api";
+import { useParams, useNavigate } from "react-router-dom";
+import UserContext from "../Context/UserContext";
+
+const AddComment = ({ setComments }) => {
+  const navigate = useNavigate();
+  const { article_id } = useParams();
+  const [error, setError] = useState(null);
+  const [disable, setDisable] = useState(0);
+  const [newComment, setNewComment] = useState("");
+  const { loggedInUser, isLoggedIn, handlePrevPath } = useContext(UserContext);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    postComment(newComment, article_id, loggedInUser.username)
+      .then((commentFromApi) => {
+        setComments((currentComments) => {
+          return [commentFromApi, ...currentComments];
+        });
+
+        setNewComment("");
+        setDisable(true);
+        setError(null);
+      })
+      .catch((error) => {
+        setError("Something went wrong, please try again.");
+      });
```

Comment:
I'm a fan of disabling the button, one thing to add though is to un-disable the button once the user can make another comment! I.e. once the previous comment has gone through and  they've written something in the text box!
Great addition of the error message to let the user know there's been a problem 👍